examples/object_detection/ios/ObjectDetection/Views/OverlayView.swift

```diff
@@ -0,0 +1,235 @@
+        .applying(
+          CGAffineTransform(
+            scaleX: maxImageDimension * offsetsAndScaleFactor.scaleFactor,
+            y: maxImageDimension * offsetsAndScaleFactor.scaleFactor)
+        )
+        .applying(
+          CGAffineTransform(
+            translationX: offsetsAndScaleFactor.xOffset - offsetToReverseModelInputTranslationX,
+            y: offsetsAndScaleFactor.yOffset - offsetToReverseModelInputTranslationY)
+        )
+
+      // Adjust the bouding boxes that extend beyond the bounds of the overlay view to fall within
+      // the overlay view.
+      if convertedRect.origin.x < 0 {
+        convertedRect.size.width = max(0, convertedRect.size.width + convertedRect.origin.x)
+        convertedRect.origin.x = 0
+      }
+
+      if convertedRect.origin.y < 0 {
+        convertedRect.size.height = max(0, convertedRect.size.height + convertedRect.origin.y)
+        convertedRect.origin.y = 0
+      }
+
+      if convertedRect.maxY > bounds.maxY {
+        convertedRect.size.height =
+          bounds.maxY - convertedRect.origin.y
+      }
+
+      if convertedRect.maxX > bounds.maxX {
+        convertedRect.size.width =
+          bounds.maxX - convertedRect.origin.x
+      }
+
+      let objectDescription = String(
+        format: "\(detection.categoryLabel) (%.2f)",
+        detection.score)
+
+      let size = objectDescription.size(withAttributes: [.font: self.displayFont])
+
+      let objectOverlay = ObjectOverlay(
+        name: objectDescription, borderRect: convertedRect, nameStringSize: size,
+        color: detection.color,
+        font: self.displayFont)
+
+      objectOverlays.append(objectOverlay)
+    }
+
+    // Hands off drawing to the OverlayView
+    self.draw(objectOverlays: objectOverlays)
+  }
+
+  // MARK: Helper Functions
+    static func offsetsAndScaleFactor(
+      forImageOfSize imageSize: CGSize,
+      tobeDrawnInViewOfSize viewSize: CGSize,
+      withContentMode contentMode: UIView.ContentMode
+    )
+      -> (xOffset: CGFloat, yOffset: CGFloat, scaleFactor: Double)
+    {
+
+      let widthScale = viewSize.width / imageSize.width
+      let heightScale = viewSize.height / imageSize.height
+
+      var scaleFactor = 0.0
+
+      switch contentMode {
+      case .scaleAspectFill:
+        scaleFactor = max(widthScale, heightScale)
+      case .scaleAspectFit:
+        scaleFactor = min(widthScale, heightScale)
+      default:
+        scaleFactor = 1.0
+      }
+
+      let scaledSize = CGSize(
+        width: imageSize.width * scaleFactor,
+        height: imageSize.height * scaleFactor)
+      let xOffset = (viewSize.width - scaledSize.width) / 2
+      let yOffset = (viewSize.height - scaledSize.height) / 2
+
+      return (xOffset, yOffset, scaleFactor)
+    }
+
+  /** Calls methods to update overlay view with detected bounding boxes and class names.
+   */
+  func draw(objectOverlays: [ObjectOverlay]) {
+
+    self.objectOverlays = objectOverlays
```

Comment:
@st-tuanmai You should clear the object overlays here. Sometimes the bounding boxes of the previous frame are visible. 

You can check the MediaPipe code for clearing the bounding boxes. Just add this function. Some variables are not defined in this overlay view. So you can remove those and just retain the lines for clearing the bounding boxes.
https://github.com/google-ai-edge/mediapipe-samples/blob/main/examples/object_detection/ios/ObjectDetector/Views/OverlayView.swift#L86

And then you can call it from this method. You can also check this method needs to be called from somewhere else. Like refer to the MediaPipe sample.